### PR TITLE
fix #19580; add warning for bare except: clause

### LINKED ISSUE #19580: Add warning for bare `except:` clause
Bare `except:` clauses have more or less undefined behavior, due to changes between Nim versions and panic dialects that cause problems for libraries - in particular, it's difficult to write library code when the user switches panic on or off and get predictable behavior: in one mode, bugs are swallowed / hidden along with regular exceptions and in the other, there's a crash.
Over time, this contributes to poor library quality.

A better alternative with the present split in exception types is being explicit about `except Defect` and `except CatchableError`, this avoids any confusion - thus, introduce a new warning `BareExcept`, enabled by default, that triggers for:

* no exception given in `except`
* any exception deriving from `Exception` but _not_ `Defect` or `CatchableError`

~~A similar warning should be added for exception types that don't inherit from one of `Decect` or `CatchableError`~~ _already exists (`InheritFromException`)_

### PR DESCRIPTION
**Pull request by ringabout**
- [x] no exception given in except

### CODE REVIEW

File: doc/manual.rst [hunk 1]
```diff
@@ -4529,8 +4529,10 @@ Example:
       echo "overflow!"
     except ValueError, IOError:
       echo "catch multiple exceptions!"
-    except:
-      echo "Unknown exception!"
+    except CatchableError:
+      echo "Catchable exception!"
+    except Defect:
+      echo "Defect!
```
Comment (konsumlamm) [inline doc/manual.rst]:
```suggestion
      echo "Defect!"
```

Comment (ringabout) [reply doc/manual.rst]: thanks!

Comment (juancarlospaco) [reply doc/manual.rst]: You can add a ` :test: ` below the `.. code-block:: nim` line to compile the code,
so this bug can not happen again, see others in the same file. 🙂

Comment (ringabout) [reply doc/manual.rst]: > You can add a :test: below the .. code-block:: nim line to compile the code,
so this bug can not happen again, see others in the same file. 🙂

You said it! Thanks!

File: doc/manual.rst [hunk 2]
```diff
@@ -4562,18 +4566,22 @@ needs to fit the types of `except` branches, but the type of the `finally`
 branch always has to be `void`:
 
 .. code-block:: nim
+    :test: "nim c $1"
   from std/strutils import parseInt
 
   let x = try: parseInt("133a")
-          except: -1
+          except CatchableError, Defect: -1
```
Comment (Menduist) [inline doc/manual.rst]: Might be a good opportunity to show better patterns, eg
`except CatchableError:`
or even
`except ValueError:`
which is is only possible of raise of parseInt AFAICT

**Pull request closed without merging**